logstash-core/lib/logstash/environment.rb

```diff
@@ -46,7 +46,11 @@ module Environment
 
   ].each {|setting| SETTINGS.register(setting) }
 
-  SETTINGS.register Setting::WritableDirectory.new("path.queue", ::File.join(SETTINGS.get("path.data"), "queue"))
+  queue_file_path = ::File.join(SETTINGS.get("path.data"), "queue")
+  # Here we have to create the directory even though it might be overridden by settings.yml
+  # We should consider a Setting::LazyDirectory class that creates the directory on first 'get'
+  FileUtils.mkdir_p([queue_file_path])
```

Comment:
@colinsurprenant I've added directory creation to the `value` method of WritableDirectory. I think this was the right place (also the comment in the code here noted that the getter was the right place to do mkdir also)